db/collection.go

```diff
@@ -167,6 +169,55 @@ func (db *db) createCollection(
 	return db.getCollectionByID(ctx, txn, desc.ID)
 }
 
+func (db *db) validateCollectionDefinitionPolicyDesc(
+	ctx context.Context,
+	policyDesc immutable.Option[client.PolicyDescription],
+) error {
+	if !policyDesc.HasValue() {
+		// No policy validation needed, weather acp module exists or not doesn't matter.
+		return nil
+	}
+
+	// If there is a policy specified, but the database does not have
+	// an acp module return an error, database must have an acp module
+	// to enable access control (inorder to adhere to the policy specified).
+	if !db.acp.HasValue() {
+		return ErrCanNotHavePolicyWithoutACPModule
+	}
+
+	// If we have the policy specified on the collection, and acp module exists,
+	// then using the acp module we need to ensure the policy id specified
+	// actually exists as a policy, and the resource name exists on that policy
+	// and that the resource is a valid DPI.
+	return db.acp.Value().ValidateResourceExistsOnValidDPI(
+		ctx,
+		policyDesc.Value().ID,
+		policyDesc.Value().ResourceName,
+	)
+}
+
+// ValidateCollectionDefinitions validates that the definitions are valid, beyond syntax.
+//
+// This is not only limited to the syntax, but also if the information
+// within the definition makes sense, this function might also make relevant
+// remote calls, like call the acp module for policy validation, and etc.
+//
+// Note: This should generally be called after parsing, i.e. on the
+// collection definitions gotten from calling `db.parser.ParseSDL`.
+func (db *db) ValidateCollectionDefinitions(
```

Comment:
todo: Only the Policy is being validated, no collection description or schema validation is done here.  Please rename this to `validatePolicyDesc` and only take a PolicyDescription.

Reply:
I can do that, I was thinking if other validation methods should also be in a similar place ( I think there are some index validation stuff that happen too ). But the change as is stands is definitely not desired. Open to better suggestion too if you want to have all validations in the same place somehow, let me know

- [ ] Will change

Reply:
CollectionDefinition is a read-only type, nothign should be trying to mutate it, and nothing should ever need to validate it.

Reply:
It isn't the collection per say that is being validated but that what the directives hold on the schema is valid (past the syntax) i.e. `@index` or `@policy`.

There is a function in file `db/collection_index.go` called `	validateIndexDescription(desc)` which runs in `db.createCollection` right after `description.SaveCollection`

Reply:
The directives you mentioned are only ever on the collection, never the schema.  IIRC the only schema directive we have is `@crdtType` (technically `@primary` but that is just historical stuff that hasn't been moved yet really).

Long-term I don't even want `CollectionDescription` and `SchemaDescription` to be properties on `CollectionDefinition`

Reply:
- [ ] Rename to `validatePolicyDesc`

Reply:
-[x] This func was completely removed and the other method is directly called in `addSchema`